### PR TITLE
[Doc Feature][Call recording API][3099832]

### PR DESCRIPTION
**Pull request by v-sdhakshina**

### CODE REVIEW

**Comment by JarbasHorst** [comment]
Hi @v-sdhakshina, can you please share with me the work item ID for this PR? Thanks!

**Comment by v-sdhakshina** [comment]
Hi @JarbasHorst , here is the work item for this PR https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/1761723.

Also, request you to please provide the response code for **Get a callRecording content** https://review.learn.microsoft.com/en-us/graph/api/callrecording-get?view=graph-rest-beta&branch=pr-en-us-21997&tabs=http#response-2

File: changelog/Microsoft.Skype.Calling.json [hunk 1]
```diff
@@ -3206,6 +3206,32 @@
       "CreatedDateTime": "2023-05-01T00:17:48.4795689Z",
       "WorkloadArea": "Teamwork and communications",
       "SubArea": "Calls and online meetings"
+    },
+    {
+      "ChangeList": [
+        {
+          "Id": "ab71be33-ee9d-4a1e-bf06-0133911ad680",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Skype.Calling.json:3213]: According to the api.md file, "The broadcastRecording property is required as we are renaming recording to broadcastRecording, recording property will be deleted later." Does this mean that onlineMeeting will have two properties "recording" and "broadcastRecording" at the same time OR that we will simple rename "recording" with "broadcastRecording"? In any case, please include this change in the onlineMeeting resource topic.

**Comment by JarbasHorst** [comment]
Hello @v-sdhakshina, I reviewed your PR and left some comments. Please, take a look when you will have time.

To avoid merge conflicts, please merge PR [22044](https://github.com/microsoftgraph/microsoft-graph-docs/pull/22044) into your branch to include my fixes and before addressing my comments. Also, please resolve the existing merge conflicts in your PR. Thanks!

File: api-reference/beta/api/callrecording-get.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+{
+    "@odata.context": "https://graph.microsoft.com/beta/$metadata#users('b935e675-5e67-48b9-8d45-249d5f88e964')/onlineMeetings('MSpiOTM1ZTY3NS01ZTY3LTQ4YjktOGQ0NS0yNDlkNWY4OGU5NjQqMCoqMTk6bWVldGluZ19ZbU0zTnpJNU9USXRZakU0WlMwME1tUTNMVGt6TVRRdFkyWm1PRGRtWmpsaVptRTNAdGhyZWFkLnYy')/recordings/$entity",
+    "id": "7e31db25-bc6e-4fd8-96c7-e01264e9b6fc",
+    "createdDateTime": "2023-04-10T08:13:17.5990966Z"
+}
+```
+
+### Example 2: Get a callRecording content
+
+> [!NOTE]
```
[the remark this answers is not in the document]
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md]: Removed note from example 2 as suggested

File: api-reference/beta/api/callrecording-get.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+GET /users/{userId}/onlineMeetings/{meetingId}/recordings/{recordingId}
+```
+
+Get the content of a single recording of an online meeting.
+
+```http
+GET me/onlineMeetings/{meetingId}/recordings/{recordingId}/content
+GET users/{userId}/onlineMeetings/{meetingId}/recordings/{recordingId}/content
+```
+
```
[the remark this answers is not in the document]
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md:57]: Added note as "This API doesn't support the optional query parameters."

File: api-reference/beta/api/callrecording-get.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+---
+
+#### Response
+
+Response contains bytes for the recording in the body. `content-type` header specifies type of the recording content. Negative offsets indicate that the recording on began while the conversation was ongoing.
+
+>**Note:** The response object shown here might be shortened for readability.
+
+```http
+
```
[the remark this answers is not in the document]
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md:141]: @JarbasHorst , the response for this request is binary stream and we need your help to update this response in the doc. Request you to please update this from your end.

File: changelog/Microsoft.Skype.Calling.json (hunk 1 above)
Comment (v-sdhakshina) [reply changelog/Microsoft.Skype.Calling.json]: Removed the changelog entry as suggested by feature PM

Comment (JarbasHorst) [reply changelog/Microsoft.Skype.Calling.json:3213]: broadcastRecording already appears in the live schema for onlineMeeting. If we go ahead and document this property now, we will be aligned with the live schema and api.md file OR will this property be introduced via another PR in the near future?

image

**Comment by JarbasHorst** [comment]
Hi @v-sdhakshina, thanks for addressing my previous comments. I left two additional comments. Please, take a look when you will have time.

Also, to avoid merge conflicts, please merge PR [22094](https://github.com/microsoftgraph/microsoft-graph-docs/pull/22094) into your branch to include my fixes and before addressing my new comments.

File: api-reference/beta/api/callrecording-get.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+> [!NOTE]
+> This API works for a meeting only if the meeting has not expired. For more details, see [Limits and specifications for Microsoft Teams](/microsoftteams/limits-specifications-teams#meeting-expiration).
+
+## HTTP request
+
+Get a single recording of an online meeting.
+
+```http
+GET /me/onlineMeetings/{meetingId}/recordings/recording/{recordingId}
```
[the remark this answers is not in the document]
Comment (devjha-ms) [reply api-reference/beta/api/callrecording-get.md:48]: The API is "/me/onlineMeetings/{meetingId}/recording/{recordingId}"

File: api-reference/beta/api/callrecording-get.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (devjha-ms) [reply api-reference/beta/api/callrecording-get.md:144]: Please update with below contentType. Other values are corerct.
Content-Type: video/mp4

File: changelog/Microsoft.Skype.Calling.json (hunk 1 above)
Comment (devjha-ms) [reply changelog/Microsoft.Skype.Calling.json:3213]: broadcastRecording property population is not yet implemented. It will be done later.

File: api-reference/beta/api/callrecording-get.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JarbasHorst) [reply api-reference/beta/api/callrecording-get.md:48]: @devjha-ms, all done, please merge PR [22094](https://github.com/microsoftgraph/microsoft-graph-docs/pull/22094) into your branch to include my fixes.

Comment (devjha-ms) [inline api-reference/beta/api/callrecording-get.md:48]: This looks like an extra line which can be removed

Comment (JarbasHorst) [reply api-reference/beta/api/callrecording-get.md:48]: @devjha-ms, can you please clarify with @v-sdhakshina, so he can apply the required changes to his PR. Thanks :)

File: api-reference/beta/api/callrecording-get.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+``` http
+GET https://graph.microsoft.com/beta/users/b935e675-5e67-48b9-8d45-249d5f88e964/onlineMeetings/MSpiOTM1ZTY3NS01ZTY3LTQ4YjktOGQ0NS0yNDlkNWY4OGU5NjQqMCoqMTk6bWVldGluZ19ZbU0zTnpJNU9USXRZakU0WlMwME1tUTNMVGt6TVRRdFkyWm1PRGRtWmpsaVptRTNAdGhyZWFkLnYy/recordings/7e31db25-bc6e-4fd8-96c7-e01264e9b6fc/content
+```
+
+#### Response
+
+The following example contains bytes for the recording in the response body. The `content-type` header specifies the type of the recording content. The negative offsets indicate that the recording began while the conversation was ongoing.
+
+> **Note:** The response object shown here might be shortened for readability.
+
```
Comment (JarbasHorst) [inline api-reference/beta/api/callrecording-get.md:142]: Hi @v-sdhakshina, looks like something is still missing here. Please add the following HTML comment before line 143.

```
<!-- {
  "blockType": "response",
  "truncated": true
}
-->
```

File: api-reference/beta/api/callrecording-get.md (hunk 5 above)
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md]: Removed "recording"

[the remark this answers is not in the document]
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md]: Merged PR 22094 as suggested

**Comment by JarbasHorst** [comment]
@v-sdhakshina, can you please try adding the following @odata.type to the HTML comment:

```
<!-- {
  "blockType": "response",
  "truncated": true, 
  "@odata.type": "stream"
}
-->
```

File: api-reference/beta/api/callrecording-get.md (hunk 6 above)
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md]: Updated as suggested

**Comment by JarbasHorst** [comment]
Congrats, @v-sdhakshina! API Doctor is validating your PR correctly now :)

**Comment by v-sdhakshina** [comment]
Thank you @JarbasHorst , request you to please approve the PR.

**Comment by JarbasHorst** [comment]
Hi @Lauragra, this PR is ready for final review. @v-sdhakshina, my colleague, Laura Graham, will take a final look at your PR before she can approve it.

**Review by devjha-ms — APPROVED** [review]
Changes looks good. Thanks!

File: api-reference/beta/api/callrecording-get.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+| Permission type                        | Permissions (from least to most privileged)                        |
+|:---------------------------------------|:-------------------------------------------------------------------|
+| Delegated (work or school account)     | OnlineMeetingRecording.Read.All                                    |
+| Delegated (personal Microsoft account) | Not supported.                                                     |
+| Application                            | OnlineMeetingRecording.Read.All, OnlineMeetingRecording.Read.Chat* |
+
+> **Note:**
+>
+> * In delegated permission, [get a callRecording content](#example-2-get-a-callrecording-content) API is supported only for meeting organizer. For more information, see [permissions or role-based access](/microsoftteams/tmr-meeting-recording-change).
```
Comment (ayan-ms) [inline api-reference/beta/api/callrecording-get.md:37]: @v-sdhakshina , for the Note related to delegated permission, please reword it to: "In delegated permission, [get a callRecording content] API is supported only for meeting organizer. Meeting participants don't have permission to download meeting recording (https://review.learn.microsoft.com/en-us/microsoftteams/tmr-meeting-recording-change?branch=main&branchFallbackFrom=pr-en-us-21997#permissions-or-role-based-access). Tenant admins can unblock meeting participants to download meeting recording. For more information, see https://learn.microsoft.com/en-us/MicrosoftTeams/block-download-meeting-recording?bc=%2FSharePoint%2Ftoc.json&toc=%2FSharePoint%2Ftoc.json"

Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md]: Updated as suggested

File: api-reference/beta/api/callrecording-get.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+| Permission type                        | Permissions (from least to most privileged)                        |
+|:---------------------------------------|:-------------------------------------------------------------------|
+| Delegated (work or school account)     | OnlineMeetingRecording.Read.All                                    |
+| Delegated (personal Microsoft account) | Not supported.                                                     |
+| Application                            | OnlineMeetingRecording.Read.All, OnlineMeetingRecording.Read.Chat* |
+
+> **Notes:**
+>
+> * In delegated permission scenarios, [getting callRecording content](#example-2-get-a-callrecording-content)  is supported only for the meeting organizer. Meeting participants don't have permission to download meeting recordings. For more information, see [permissions or role-based access](/microsoftteams/tmr-meeting-recording-change). Tenant admins can unblock meeting participants to download meeting recording. For more information, see [block the download of Teams meeting recording files](/MicrosoftTeams/block-download-meeting-recording).
```
Comment (Lauragra) [inline api-reference/beta/api/callrecording-get.md:37]: Please clarify the second sentence. It seems like you are saying that the admin can "unblock" meeting participants - does that mean the admin can give them the ability to download the meeting recording? The article you link to is about blocking, not unblocking, so that is unclear.

File: api-reference/beta/api/callrecording-get.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+> * Permissions marked with * use [resource-specific consent](/microsoftteams/platform/graph-api/rsc/resource-specific-consent).
```
Comment (Lauragra) [inline api-reference/beta/api/callrecording-get.md:38]: Note that when we use an asterisk in permission tables, the table has to be excluded from automation. In the near term, we will be automating the permissions tables so they will no longer be hand-authored, except in cases where there are asterisks, superscripts, and the like associated with notes. In this case, because we only have one asterisk, it might be okay to just name the permission in the note; i.e. "The application permission  OnlineMeetingRecording.Read.Chat uses... ".

File: api-reference/beta/api/onlinemeeting-list-recordings.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged)                        |
+|:---------------------------------------|:-------------------------------------------------------------------|
+| Delegated (work or school account)     | OnlineMeetingRecording.Read.All                                    |
+| Delegated (personal Microsoft account) | Not supported.                                                     |
+| Application                            | OnlineMeetingRecording.Read.All, OnlineMeetingRecording.Read.Chat* |
+
+> **Note:** Permissions marked with * use [resource-specific consent](/microsoftteams/platform/graph-api/rsc/resource-specific-consent).
```
Comment (Lauragra) [inline api-reference/beta/api/onlinemeeting-list-recordings.md:31]: Note my previous comment regarding asterisks in permission tables.

**Review by Lauragra — APPROVED** [review]

File: api-reference/beta/api/callrecording-get.md (hunk 9 above)
Comment (v-sdhakshina) [reply api-reference/beta/api/callrecording-get.md:38]: Updated as suggested

File: api-reference/beta/api/onlinemeeting-list-recordings.md (hunk 10 above)
Comment (v-sdhakshina) [reply api-reference/beta/api/onlinemeeting-list-recordings.md:31]: Updated as suggested

**Comment by v-sdhakshina** [comment]
Hi @Lauragra , request you to please merge the PR to live by 9th August 2023.
